url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/world/Level.java

```diff
@@ -0,0 +1,424 @@
+	 * To do this, use {@link Player#GlobalBlockChange(short, short, short, Block, Level, Server)} instead
+	 * @param b
+	 *         The block to add
+	 * @param x
+	 *        The X coordinate
+	 * @param y
+	 *        The Y coordinate
+	 * @param z
+	 *        The Z coordinate
+	 * @param server
+	 *              The server this blockchange is happening in
+	 */
+	public void setTile(Block b, int x, int y, int z, Server server) {
+		setTile(b, PosToInt(x, y, z), server);
+	}
+	
+	/**
+	 * Convert coordinates to a number that will
+	 * correspond to where the coordinates are in the
+	 * block array
+     * @param x
+	 *        The X coordinate
+	 * @param y
+	 *        The Y coordinate
+	 * @param z
+	 *        The Z coordinate
+	 * @return
+	 *        The number that will correspond to where the coordinates
+	 *        are in the block array
+	 */
+	public int PosToInt(int x, int y, int z) {
+        if (x < 0) { return -1; }
+        if (x >= width) { return -1; }
+        if (y < 0) { return -1; }
+        if (y >= height) { return -1; }
+        if (z < 0) { return -1; }
+        if (z >= depth) { return -1; }
+        return x + z * width + y * width * depth;
+    }
+	
+	public int[] IntToPos(int index) {
+		int[] toreturn = new int[3];
+		toreturn[1] = (index / width / height);
+		index -= toreturn[1]*width*height;
+		toreturn[2] = (index/width);
+		index -= toreturn[2]*width;
+		toreturn[0] = index;
+		return toreturn;
+	}
+	
+	/**
+	 * Save the level
+	 * @throws IOException
+	 *                   An IOExceptoin is thrown if there is a problem writing to the file
+	 */
+	public void Save() throws IOException {
+		if (!new File("levels").exists())
+			new File("levels").mkdir();
+		FileOutputStream fos = new FileOutputStream("levels/" + name + ".ggs");
+		GZIPOutputStream gos = new GZIPOutputStream(fos);
+		ObjectOutputStream out = new ObjectOutputStream(gos);
+		out.writeLong(serialVersionUID);
+		out.writeObject(this);
+		out.close();
+		gos.close();
+		fos.close();
+	}
+	
+	/**
+	 * Unload this level.
+	 * All players who are in this level will be sent to the {@link Server#MainLevel}
+	 * @param server
+	 *             The server thats unloading the level
+	 * @throws IOException
+	 *                   An IOException will occur if there is a problem saving the level
+	 */
+	public void Unload(Server server) throws IOException {
+		Unload(server, true);
+	}
+	/**
+	 * Unload this level.
+	 * All players who are in this level will be sent to the {@link Server#MainLevel}
+	 * @param server
+	 *             The server thats unloading the level
+	 * @param save
+	 *           Weather the level should save before unloading
+	 * @throws IOException
+	 *                   An IOException will occur if there is a problem saving the level
+	 */
+	public void Unload(Server server, boolean save) throws IOException {
+		if (save)
+			Save();
+		run = false;
+		try {
+			physics.join();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)